src/interactor/state_machine/interactor_steps/AddLabelStep.cc

```diff
@@ -0,0 +1,90 @@
+#include "AddLabelStep.h"
+
+#include "interactor/io_facility/Strings.h"
+#include "interactor/state_machine/interactor_steps/PromptStep.h"
+#include "utils/TaskIdUtils.h"
+
+namespace task_manager {
+
+std::unique_ptr<Command> AddLabelStep::execute(Context ctx) {
+  --stage_;
+  if (stage_ == 1) {
+    return HandleStage<1>(ctx);
+  }
+  if (stage_ == 0) {
+    return HandleStage<0>(ctx);
+  }
+  std::terminate();
+}
+
+template <>
+std::unique_ptr<Command> AddLabelStep::HandleStage<1>(Context &) {
+  if (arg_.empty()) {
+    return ReportError(Strings::kRequiredId);
+  }
+
+  auto token = validator_->ConsumeOneTokenFrom(arg_);
+  auto to_edit = validator_->ParseInt(token);
+  if (!to_edit) {
+    return ReportError(Strings::InvalidId(token));
+  }
+  task_id_.set_id(*to_edit);
+
+  if (arg_.empty()) {
+    return ReportError(Strings::kRequiredLabel);
+  }
+
+  token = validator_->ConsumeOneTokenFrom(arg_);
+  if (!arg_.empty()) {
+    return ReportError(Strings::kMultipleArgumentDoesNotSupported);
+  }
+
+  label_.set_name(token);
+  return std::make_unique<GetSpecifiedTasksCommand>(
+      std::vector<TaskId>{task_id_});
+}
+
+template <>
+std::unique_ptr<Command> AddLabelStep::HandleStage<0>(Context &ctx) {
+  if (!ctx.solid_tasks &&
+      ctx.status == ModelController::Status::kNotPresentId) {
```

Comment:
doesn't std::nullopt imply kNotPresentId automatically? other statuses are not processed

Reply:
Its an additional check, that is redundant, I agree. `!ctx.solid_tasks` check is removed.